receiver/pulsarreceiver/receiver_logs.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pulsarreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/pulsarreceiver"
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+)
+
+const alreadyClosedError = "AlreadyClosedError"
+
+type pulsarLogsReceiver struct {
+	logsConsumer consumer.Logs
+	unmarshaler  LogsUnmarshaler
+	client       pulsar.Client
+	consumer     pulsar.Consumer
+	cancel       context.CancelFunc
+	settings     receiver.CreateSettings
+	config       Config
+}
+
+func newLogsReceiver(config Config, set receiver.CreateSettings, unmarshalers map[string]LogsUnmarshaler, nextConsumer consumer.Logs) (*pulsarLogsReceiver, error) {
+	option := config.Log
+	if err := option.validate(); err != nil {
+		return nil, err
+	}
+	unmarshaler := unmarshalers[option.Encoding]
+	if nil == unmarshaler {
```

Comment:
```suggestion
	if unmarshaler == nil {
```
I think we should place variables to the left-hand side in control statements, to be consistent with standards.